### DISCUSSION TITLE
Add more matrixes?

### DISCUSSION THREAD

**Discussion by pinterf** [pinterf/assrender#5] [Ideas]
tv.601 and tv.709 is handled already.
Plugin accepts additionally bt.2020 or Rec2020.

Does it have any reason to implement the rest?
"YCbCr Matrix" can be "tv.601" "pc.601" "tv.709" "pc.709" "tv.240m" "pc.240m" "tv.fcc" "pc.fcc"

**Comment by realfinder** [comment]
xy-VSFilter has "None" too (also Aegisub has it)

> [Script Info]
> YCbCr Matrix: TV.601
> YCbCr Matrix: TV.709
> YCbCr Matrix: PC.601
> YCbCr Matrix: PC.709
> YCbCr Matrix: None
> 'YCbCr Matrix' as well as override settings in the GUI, only have an effect on YCbCr subtitle output. The video is always untouched.
> 
> TV.601 = Output BT.601 with 16-235 YCbCr level range
> TV.709 = Output BT.709 with 16-235 YCbCr level range
> PC.601 = Output BT.601 with 0-255 YCbCr level range
> PC.709 = Output BT.709 with 0-255 YCbCr level range
> None = Output BT.601 or BT.709 with 16-235 YCbCr level range based on video resolution
> 
>  If 'YCbCr Matrix' is missing from a script, xy-VSFilter will use TV.601 for legacy script compatibility. It is highly recommended to leave the GUI settings at their defaults of Auto | Auto for 'YCbCr level range' | 'YCbCr matrix'. The Auto settings obey the 'YCbCr Matrix' tag (if present), else fallback to TV.601 as decribed above.

**Reply by pinterf** [reply]
So explicite "none" is doing the auto-guess?
(meanwhile pc.601, pc.709, tv.fcc, pc.fcc, tv.240m and pc.240m are implemented)

**Reply by realfinder** [reply]
yes, it's like the auto-guess that assrender was do

**Comment by pinterf** [comment]
And why is bt.2020 not mentioned for sources larger than fhd? Or is it only for 10-12 bits?

**Reply by realfinder** [reply]
I have no idea, I didn't have any UHD or HDR source, but yes I think it's only for 10-12 bits

**Reply by realfinder** [reply]
seems bt.2020 is not used yet in ssa/ass https://github.com/libass/libass/issues/297

so I think it's better to be excluded from "None" auto-guess at least for now (until [Aegisub](https://github.com/Aegisub/Aegisub) added)

**Reply by pinterf** [reply]
Long discussion but with zero result. However someone has put it on request six days ago.
